crates/bevy_transform/src/systems.rs

```diff
@@ -10,16 +11,29 @@ use bevy_hierarchy::{Children, Parent};
 /// Third party plugins should use [`transform_propagate_system_set`](crate::transform_propagate_system_set)
 /// to propagate transforms correctly.
 pub fn sync_simple_transforms(
-    mut query: Query<
-        (&Transform, &mut GlobalTransform),
-        (Changed<Transform>, Without<Parent>, Without<Children>),
-    >,
+    mut query: ParamSet<(
+        Query<
+            (&Transform, &mut GlobalTransform),
+            (Changed<Transform>, Without<Parent>, Without<Children>),
+        >,
+        Query<(Ref<Transform>, &mut GlobalTransform), Without<Children>>,
+    )>,
+    orphaned: RemovedComponents<Parent>,
 ) {
     query
+        .p0()
         .par_iter_mut()
         .for_each_mut(|(transform, mut global_transform)| {
             *global_transform = GlobalTransform::from(*transform);
         });
+    // updated orphaned entities
+    let mut query = query.p1();
+    let mut iter = query.iter_many_mut(orphaned.iter());
+    while let Some((transform, mut global_transform)) = iter.fetch_next() {
```

Comment:
```suggestion
    for (transform, mut global_transform) in iter {
```
Why are we using `fetch_next` over `Iterator::next` here?

Reply:
It's because `QueryManyIter` only implements `Iterator` if the items are `ReadOnly`, and here we need mutable access to `GlobalTransform`. The `iter_many_mut` kinda implies that without saying why https://dev-docs.bevyengine.org/bevy/ecs/system/struct.Query.html#method.iter_many_mut

Check out `impl QueryManyIter` and `impl Iterator for QueryManyIter` for an explanation. Not sure the restriction is actually correctly justified, but I think that would out of scope for this PR.

Reply:
If the list of entities given to `iter_many_mut` contains duplicate entities then an `Iterator` could mutably alias (eg. using `collect()`), so we return our own 'lending' iterator instead that only allows access to one item at a time